File: tutorials/testing/c++/tut_shapes_p1.cpp
#include "Bridges.h"
#include "SymbolCollection.h"
#include "SymbolGroup.h"
#include "Rectangle.h"
#include "Polyline.h"
#include "Polygon.h"
#include "Circle.h"
#include "Text.h"

using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// title, description
	bridges.setTitle("Symbol Collection: Display shapes supported by BRIDGES");


	// create some symbols and add to symbol collection
	SymbolCollection sc;

	Rectangle r1(50, 0, 25, 25);

	r1.setFillColor(Color("red"));
		r1.setOpacity(0.8f);
		sc.addSymbol(r1);
	sc.addSymbol(r1);
	
	Rectangle r2(-50, 0, 25, 25);
		r2.setFillColor(Color("orange"));
		r2.setOpacity(0.4f);
	sc.addSymbol(r2);

	// and a circle next to it
	Circle c1(50, 50, 15);
		c1.setFillColor(Color("green"));
		c1.setOpacity(1.0f);
	sc.addSymbol(c1);

	// add the group in the symbol collection 
	SymbolCollection sc2;

	Polygon p1;
		p1.addPoint(-25, -10);
		p1.addPoint(25, -10);
		p1.addPoint(0, 50);
		p1.setStrokeColor("darkorchid");
		p1.setStrokeWidth(2.0f);
		p1.setOpacity(1.0f);
	sc.addSymbol(p1);

	// draw axes
	Polyline p3;
		p3.addPoint(-100, 0);
		p3.addPoint(100, 0);
		p3.setStrokeColor("magenta");
		p3.setStrokeWidth(2.0f);
		p3.setOpacity(1.0f);
	sc.addSymbol(p3);

	Polyline p2;
		p2.addPoint(0, -100);
		p2.addPoint(0, 100);
		p2.setStrokeColor("magenta");
		p2.setStrokeWidth(2.0f);
		p2.setOpacity(1.0f);
	sc.addSymbol(p2);


	// an example text label
	Text l;
	float loc[2] = {10, 25};
		l.setAnchorLocation(loc);
		l.setFontSize(20.0f);
		l.setStrokeWidth(1.0f);
		l.setText("test label");
		l.setStrokeColor("purple");
		l.setOpacity(1.0f);
	sc.addSymbol(l);

	// set visualizer type
	bridges.setDataStructure(sc);

	// visualize the JSON and Collection
	bridges.visualize();

	return 0;
}

File: tutorials/testing/java/tut_shapes_p1.java
import bridges.connect.Bridges;
import bridges.base.SymbolCollection;
import bridges.base.Rectangle;
import bridges.base.Circle;
import bridges.base.Text;
import bridges.base.Polyline;
import bridges.base.Polygon;
import bridges.base.Color;
import bridges.base.SymbolGroup;


public class tut_shapes_p1 {

	public static void main(String[] args) throws Exception {

		//create Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");
		// title, description
		bridges.setTitle("Symbol Collection");
		bridges.setDescription("Red square, green circle, magenta horizontal and vertical lines, "
			+	"and a test label with a purple outline.");

		// create some symbols and add to symbol collection
		SymbolCollection sc = new SymbolCollection();

		// draw a rectangle
		Rectangle s1 = new Rectangle(-50.0f, 0.0f, 25.0f, 25.0f);
			s1.setFillColor("red");
			s1.setOpacity(0.3f);
		sc.addSymbol(s1);

		// draw a rectangle
		Rectangle r1 = new Rectangle(50.0f, 0.0f, 25.0f, 25.0f);
			r1.setFillColor("orange");
			r1.setOpacity(0.8f);
		sc.addSymbol(r1);

		// and a circle next to it
		Circle s2 = new Circle(50.0f, 50.0f, 15.0f);
			s2.setFillColor("green");
			s1.setOpacity(0.6f);
		sc.addSymbol(s2);

		// draw a triangle above
		Polygon s3 = new Polygon();
			s3.addPoint(-25.0f, -10.0f);
			s3.addPoint(25.0f, -10.0f);
			s3.addPoint(0.0f, 50.0f);
			s3.setStrokeColor("darkorchid");
			s3.setStrokeWidth(2.0f);
		sc.addSymbol(s3);

		// draw axes
		Polyline s4 = new Polyline();
			s4.addPoint(-100.0f, 0.0f);
			s4.addPoint(100.0f, 0.0f);
			s4.setStrokeColor("magenta");
			s4.setStrokeWidth(2.0f);
		sc.addSymbol(s4);

		Polyline s5 = new Polyline();
			s5.addPoint(0.0f, -100.0f);
			s5.addPoint(0.0f, 100.0f);
			s5.setStrokeWidth(2.0f);
			s5.setStrokeColor("magenta");
		sc.addSymbol(s5);

		Text l = new Text();
		l.setAnchorLocation(0.0f, 25.0f);
		l.setFontSize(18f);
		l.setStrokeWidth(2.0f);
		l.setOpacity(.3f);
		l.setText("test label1");
		l.setStrokeColor("purple");
		sc.addSymbol(l);


		// set visualizer type
		bridges.setDataStructure(sc);

		// visualize 
		bridges.visualize();
	}
}

